adcc/AdcMatrix.py

```diff
@@ -32,6 +32,48 @@
 from .AmplitudeVector import AmplitudeVector
 
 
+class AdcExtraTerm:
+    # NOTE: currently requires the matrix
+    # to allow for block construction with the same
+    # interface as the usual adc_pp, i.e.,
+    # reference_state, ground_state, intermediates
+    def __init__(self, matrix, blocks):
```

Comment:
Why do you need to pass the matrix? If you pass ground state it should be sufficient, shouldn't it? (ok save the intermediates, but I think we should have an interface that does not require the matrix as well. It somehow feels "too much".

Reply:
I had the same thought in the beginning, but after having worked with it a couple of times now, it somehow feels right: "Why would you construct an extra term without having a matrix that it belongs to in the first place?" is my current take on this issue.

If you think it's too much, we can change the interface, but then we need to adapt the underlying block construction routine accordingly (which is then inconsistent wrt the normal adc_pp routines).

Reply:
ok I see. I have not checked that closely. Let's leave it and consider the other aspects first.